feat: support candlestick chart

nk46-cloud opened the pull request:
<!-- Suggested PR Title: [feat/fix/refactor/perf/test/ci/docs/chore] brief description of the change -->
<!-- Please follow Conventional Commits: https://www.conventionalcommits.org/en/v1.0.0/ -->

## Description
This PR adds support for candlestick plot using maidr-ts and mplfinance library.

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

nk46-cloud commented on maidr/core/plot/candlestick.py:
```diff
@@ -0,0 +1,239 @@
+                    "high": high_price,
+                    "low": low_price,
+                    "close": close_price,
+                    "volume": 0,
```
Business logic not defined for volume yet

jooyoungseo reviewed (changes requested):
How about we fetch volume and provide it as an additional line layer?

nk46-cloud commented:
> How about we fetch volume and provide it as an additional line layer?

I have added an example where volume is shown and added as a separate layer in the graph. In the legacy version (from mplfinance.original_flavor import candlestick_ohlc) of  this function, we have to add a separate axis since there is no support within that function. However, as per documentation the newer version can add volume lines easily with extra paramater. Example:     import mplfinance as mpf mpf.plot(daily,type='candle',mav=(3,6,9),volume=True) 

We have to change the patch logic completely. Please let me know if we can add support for this additional function as well. I can open an issue to track this. Docs of mpl finance for your reference: https://github.com/matplotlib/mplfinance?tab=readme-ov-file Thanks!

nk46-cloud commented:
> @nk46-cloud Great! Yes, let's refactor candlestick patch to support the new param. Feel free to file an issue and make a separate PR. We can merge this PR first.

Thank you. I am working on the github actions issue. I don't see release wf being triggered even if pr title follow conventional commit and includes feat keyword. I'll merge this once I find the fix.
